fortuna/prob_model/posterior/sgmcmc/sghmc/sghmc_posterior.py

```diff
@@ -0,0 +1,202 @@
+import logging
+from typing import Optional
+from itertools import cycle
+import pathlib
+
+from fortuna.data.loader import DataLoader
+from fortuna.prob_model.fit_config.base import FitConfig
+from fortuna.prob_model.joint.base import Joint
+from fortuna.prob_model.posterior.map.map_trainer import (
+    MAPTrainer,
+    JittedMAPTrainer,
+    MultiDeviceMAPTrainer,
+)
+from fortuna.prob_model.posterior.run_preliminary_map import run_preliminary_map
+from fortuna.prob_model.posterior.map.map_posterior import MAPPosterior
+from fortuna.prob_model.posterior.map.map_state import MAPState
+from fortuna.prob_model.posterior.base import Posterior
+from fortuna.prob_model.posterior.posterior_multi_state_repository import \
+    PosteriorMultiStateRepository
+from fortuna.prob_model.posterior.sgmcmc.sgmcmc_posterior_mixin import \
+    SGMCMCPosteriorMixin
+from fortuna.prob_model.posterior.sgmcmc.sghmc import SGHMC_NAME
+from fortuna.prob_model.posterior.sgmcmc.sghmc.sghmc_approximator import \
+    SGHMCPosteriorApproximator
+from fortuna.prob_model.posterior.sgmcmc.sghmc.sghmc_callback import \
+    SGHMCSamplingCallback
+from fortuna.prob_model.posterior.sgmcmc.sghmc.sghmc_integrator import \
+    sghmc_integrator
+from fortuna.prob_model.posterior.sgmcmc.sghmc.sghmc_state import SGHMCState
+from fortuna.typing import Status
+from fortuna.utils.device import select_trainer_given_devices
+
+logger = logging.getLogger(__name__)
+
+
+class SGHMCPosterior(SGMCMCPosteriorMixin, Posterior):
+    def __init__(
+        self,
+        joint: Joint,
+        posterior_approximator: SGHMCPosteriorApproximator,
+    ):
+        """
+        Stochastic Gradient Hamiltonian Monte Carlo approximate posterior class.
+
+        Parameters
+        ----------
+        joint: Joint
+            A Joint distribution object.
+        posterior_approximator: SGHMCPosteriorApproximator
+            A SGHMC posterior approximator.
+        """
+        super().__init__(
+            joint=joint, posterior_approximator=posterior_approximator
+        )
+
+    def __str__(self):
+        return SGHMC_NAME
+
+    def fit(
+        self,
+        train_data_loader: DataLoader,
+        val_data_loader: Optional[DataLoader] = None,
+        fit_config: FitConfig = FitConfig(),
+        map_fit_config: Optional[FitConfig] = None,
```

Comment:
Here you do use the preliminary run with MAP. How come here it appears but not in Cyclical SGLD? I guess I missed something before.